prog/dftb+/lib_common/environment.F90

```diff
@@ -96,21 +105,67 @@ module environment
 
 contains
 
-  !> Returns an initialized instance.
   subroutine TEnvironment_init(this)
 
     !> Instance
     type(TEnvironment), intent(out) :: this
 
-    call TTimerArray_init(this%globalTimer, globalTimerItems)
+    call TFileRegistry_init(this%fileFinalizer)
 
   end subroutine TEnvironment_init
 
 
+  !> Finalizes the environment.
+  subroutine TEnvironment_destruct(this)
+
+    !> Instance
+    class(TEnvironment), intent(inout) :: this
+
+    call this%globalTimer%writeTimings()
+    call this%fileFinalizer%closeAll()
+    flush(stdOut)
+
+  end subroutine TEnvironment_destruct
+    
+  
+  !> Gracefully aborts finalizing the current environment first
+  subroutine TEnvironment_abort(this)
+
+    !> Instance
+    class(TEnvironment), intent(out) :: this
+
+    call this%destruct()
+    call abort()
+
+  end subroutine TEnvironment_abort
+
+
+  !> Initlaizes the global timer of the environment.
+  subroutine TEnvironment_initGlobalTimer(this, timingLevel, header, unit)
+
+    !> Instance
+    class(TEnvironment), intent(inout) :: this
+
+    !> Timing level up to which timings should be printed (-1 for all)
```

Comment:
Manual may have to be revised if -1 prints all levels.

Reply:
Done